### DISCUSSION TITLE
Advice on my find-reference addon.

### DISCUSSION THREAD

**Discussion by OriLiMu** [ScratchAddons/ScratchAddons#7590] [Discussion]
I want to make an addon about find all reference for a variable/list or method to a list in a float new window with some brief context on each reference. 

There is a `find-bar` addon I know, but the progress to look through all the reference is a little inconvenient for me, because you only can check one at a time, jump between two  **un-near** references is impossible. 

But I havent make an addon before. So I make this post to discuss with the experts. 

What I want to make is:
1. When you hover on the result of `find-bar`, there is a float window show on the right side. 
  - the float window contains a list of usage of the variable
  - if click on the item will jump to the exact usage in the code.
  - each item of list contains (left to right):
    1. start block of the blocks where the variable/list/method in.(most of should be `when flag clicked`, `when start as a clone` and `self-defined method name`)
    2. 2 nearest control command(if/repeat...). for Example, I have a code with structure, the 2 nearest control command is : repeat 10, and if y > 0
```
repeat 5
  repeat 10
    if variable y > 0:
      x += 1
```

# I want to know
1. If my design about the addon is workable? 
2. And certainly, I need to change the code of `find-bar` which is already part of the addons. I want to know if my code can be submitted to the `scratchaddons` if I change another addon code in this case.

**Comment by WorldLanguages** [comment]
If you want to submit your change to the project, ideally you should create an issue in this repository explaining the problem you want to solve, and ideally some UI mockups. Only after that code is written.
Creating an issue also means someone else can recapture your idea if you end up not submitting code.

**Reply by OriLiMu** [reply]
Can you give me a example issue which contains the UI mockup? I want to check it as a reference . And I don't know what ui mockup should be like. 

I read some issue but cant find issues with mockup

**Reply by WorldLanguages** [reply]
It's not necessary to add a UI mockup, but it could make your issue easier to read and understand.
Here is an example: #7460
